spec/models/spree/order/state_machine_spec.rb

```diff
@@ -136,21 +136,16 @@
       end
 
       context "without shipped items" do
-        it "should set payment state to 'credit owed'" do
+        it "should set payment state to 'void'" do
           order.cancel!
-          expect(order.payment_state).to eq 'credit_owed'
+          expect(order.payment_state).to eq 'void'
```

Comment:
This spec doesn't actually test if the change is persisted. That's something you fixed with update_column. I think that we should write this spec to cover that:
```suggestion
          expect {
            order.cancel!
            order.reload
          }.to change { order.payment_state }.to("void")
```

Reply:
Nice! good idea